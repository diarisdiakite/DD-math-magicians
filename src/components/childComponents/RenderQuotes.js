import React from 'react';
import PropTypes from 'prop-types';

function RenderQuotes({ quotes }) {
  return (
    <div>
      {quotes.map((quote) => (
        <div key={quote.id}>
          <p>{quote.quote}</p>
          <p>{quote.author}</p>
          <p>{quote.category}</p>
        </div>
      ))}
    </div>
  );
}

RenderQuotes.propTypes = {
  quotes: PropTypes.arrayOf(
    PropTypes.shape({
      quote: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default RenderQuotes;
