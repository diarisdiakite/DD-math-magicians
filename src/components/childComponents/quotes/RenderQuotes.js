import React from 'react';
import PropTypes from 'prop-types';

function RenderQuotes({ quotes }) {
  return (
    <div>
      {quotes.map((quote) => (
        <div key={quote.id}>
          <p>
            Your takeaway
            {' '}
            {quote.category}
            {' '}
            quote
          </p>
          <h3 key={`quote-${quote.id}`}>{quote.quote}</h3>
          <p key={`author-${quote.id}`}>{quote.author}</p>

        </div>
      ))}
    </div>
  );
}

RenderQuotes.propTypes = {
  quotes: PropTypes.arrayOf(
    PropTypes.shape({
      quote: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      category: PropTypes.string,
    }),
  ).isRequired,
};

export default RenderQuotes;
