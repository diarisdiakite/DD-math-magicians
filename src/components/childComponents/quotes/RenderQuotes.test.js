import renderer from 'react-test-renderer';
import RenderQuotes from './RenderQuotes';

describe('RenderQuotes component', () => {
  it('renders correctly', () => {
    const quotes = [
      {
        id: 1878900,
        randomCategory: 'intelligence',
        quote: 'This is a test quote',
        author: 'quote author',
      },
    ];

    const tree = renderer
      .create(
        <RenderQuotes quotes={quotes} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();

    const firstQuote = quotes[0];
    expect(firstQuote.id).toEqual(expect.any(Number));
    expect(firstQuote.author).toEqual(expect.any(String));
  });
});
