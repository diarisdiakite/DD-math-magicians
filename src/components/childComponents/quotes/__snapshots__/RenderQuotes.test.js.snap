// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderQuotes component renders correctly 1`] = `
<div>
  <div>
    <p>
      Your takeaway
       
       
      quote
    </p>
    <h3>
      This is a test quote
    </h3>
    <p>
      quote author
    </p>
  </div>
</div>
`;
